microsoft-edge/webview2/get-started/hololens2.md

```diff
@@ -0,0 +1,605 @@
+
+* [Input overview — MRTK2](/windows/mixed-reality/mrtk-unity/mrtk2/features/input/overview) - recommended for Mixed Reality Toolkit 2.8 applications.
+* [Input — MRTK3](/windows/mixed-reality/mrtk-unity/mrtk3-input/packages/input/overview) - recommended for Mixed Reality Toolkit 3 applications.
+* [Unity Input System](https://docs.unity3d.com/Packages/com.unity.inputsystem@1.5/manual/index.html)
+
+Regardless of the input system used within your Unity application, interop code between the various application input events and the WebView plugin for Unity is required. This means translating those events (such as Pointer events) into a `WebViewMouseEventData` object and then forwarding those events to the plugin via the `IWithMouseEvent` interface:
+
+```C#
+public interface IWithMouseEvents : IWithInputEvents
+{
+    void MouseEvent(WebViewMouseEventData mouseEvent);
+}
+```
+
+WebView2 is unaware of Unity's input system and likely has a different coordinate system than your Unity scene. As a result, when there is a pointer-down event, its coordinates must be translated into the coordinate system of the WebView2 control. Additionally, the pointer-down event needs to be converted into an appropriate `WebViewMouseEventData` event type.
+
+Simple example:
+
+```C#
+using Microsoft.MixedReality.WebView;
+
+public class WebViewExample : MonoBehaviour, IPointerDownHandler
+{
+    // WebView setup steps skipped for brevity
+
+    public void OnPointerDown(PointerEventData eventData)
+    {
+        IWithMouseEvents mouseEventsWebView = webView as IWithMouseEvents;
+
+        // Call hypothetical function which converts the event's x, y into the WebView2's coordinate space.
+        var hitCoord = ConvertToWebViewSpace(eventData.position.x, eventData.position.y);
+
+        WebViewMouseEventData mouseEvent = new WebViewMouseEventData
+        {
+            X = hitCoord.x,
+            Y = hitCoord.y,
+            Type = PointerEvent.PointerDown,
+            Button = PointerButton.Left,
+            TertiaryAxisDeviceType = WebViewMouseEventData.TertiaryAxisDevice.PointingDevice
+        };
+
+        // Propagate the event to the WebView plugin.
+        mouseEventsWebView.MouseEvent(mouseEvent);
+    }
+}
+```
+
+In the above example, pointer-down events are converted into `WebViewMouseEventData` objects and forwarded to the WebView plugin for Unity. It is essentially converted into a mouse-down event. In order to effect mouse click events, pointer-up events would need to be handled in a similar fashion.
+
+In the example above, `ConvertToWebViewSpace` is intentionally not implemented.
+
+
+<!-- ====================================================================== -->
+## See also
+
+* [Get started with WebView2 in WinUI 2 (UWP) apps](./winui2.md)
+* [WebView2 API Reference](../webview2-api-reference.md)
+<!-- todo: un-comment when http://learn.microsoft.com/windows/mixed-reality/develop/advanced-concepts/webview2-unity-plugin exists
```

Comment:
@michaelfarnsworth @RobertVS-MS -- should this PR un-comment this link?  There are 3 instances in this file.  

If we un-comment them in this PR, the doc build will report three 404s, until the target PR is merged to "live".  After that PR is published, we could push a whitespace or date change in the present PR, to trigger a doc build and see the 404s go away.

Reply:
I think we should go ahead and un-comment. We're planning to publish multiple related PRs, all with links pointing to each other. So, until everything's live, we're going to have broken links.

Reply:
done.  
https://github.com/michaelfarnsworth/edge-developer/blob/webview2-hololens-getting-started/microsoft-edge/webview2/get-started/hololens2.md
3x   "API Reference"

can close this thread